runtime/lua/vim/lsp.lua

```diff
@@ -364,7 +365,7 @@ do
       local curr_lines = nvim_buf_get_lines(bufnr, 0, -1, true)
       local line_ending = format_line_ending[vim.api.nvim_buf_get_option(0, 'fileformat')]
       local incremental_change = sync.compute_diff(
-        cached_buffers[bufnr], curr_lines, firstline, lastline, new_lastline, client.offset_encoding or 'utf-16', line_ending or '\n')
+        cached_buffers[bufnr], curr_lines, firstline, lastline, new_lastline, client.offset_encoding, line_ending or '\n')
```

Comment:
there's no reason to remove the fallback to utf-16, it's the default for most language servers

Reply:
The fallback should still happen at a lower level.